human-forces-provider/src/RobotFrameTransformer.cpp

```diff
@@ -0,0 +1,116 @@
+
+#include <yarp/os/Log.h>
+#include <yarp/dev/IEncoders.h>
+
+#include <iDynTree/KinDynComputations.h>
+#include <iDynTree/yarp/YARPConversions.h>
+
+
+namespace human
+{
+    RobotFrameTransformer::RobotFrameTransformer(const iDynTree::Transform &constTransformFromFixture,
+                                                 const std::string &robotArm,
+                                                 const std::string &robotSole,
+                                                 const std::string &humanFoot,
+                                                 const std::string &humanHand,
+                                                 yarp::dev::IEncoders &robotJointConfiguration_encoder,
+                                                 yarp::os::BufferedPort<HumanState> &humanJointConfiguration_port)
+    : GenericFrameTransformer(robotArm,humanHand)
+    , m_constTransformFromFixture(constTransformFromFixture)
+    , m_robotSole(robotSole)
+    , m_humanFoot(humanFoot)
+    , m_humanHand(humanHand)
+    , m_robotJointConfiguration_encoder(robotJointConfiguration_encoder)
+    , m_humanJointConfiguration_port(humanJointConfiguration_port){}
+    
+    
+    bool RobotFrameTransformer::init(const iDynTree::Model &humanModel,
+                                     const iDynTree::Model &robotModel)
+    {
+        if (!m_humanComputations.loadRobotModel(humanModel))
+        {
+            yError("Error in reading the human model!");
+            return false;
+        }
+        
+        m_humanConfiguration.resize(m_humanComputations.getNrOfDegreesOfFreedom());
+        m_humanVelocity.resize(m_humanComputations.getNrOfDegreesOfFreedom());
+        m_humanVelocity.zero();
+        
+        if (!m_robotComputations.loadRobotModel(robotModel))
+        {
+            yError("Error in reading the robot model!");
+            return false;
+        }
+        
+        m_robotConfiguration.resize(m_robotComputations.getNrOfDegreesOfFreedom());
+        m_robotVelocity.resize(m_robotComputations.getNrOfDegreesOfFreedom());
+        m_robotVelocity.zero();
+        
+        return true;
+        
+    }
+    
+    
+    bool RobotFrameTransformer::transformForceFrame(const yarp::sig::Vector &inputforce,
+                                                    yarp::sig::Vector &transformedForce,
+                                                    std::string &transformedExpressedFrame)
+    {
+        // UPDATE MATRIX
+        
+        iDynTree::Vector3 worldGravity;
+        worldGravity.zero(); worldGravity(2) = -9.81;
+
+        // -----read human configuration
+        human::HumanState *tmp = m_humanJointConfiguration_port.read();
+        
+        // transform yarp Vector in iDynTree vector
+        if (!iDynTree::toiDynTree(tmp->positions,m_humanConfiguration))
+        {
+            yError("Somenthing wrong in the convertion from YARP to iDynTree vector!");
+            return false;
+        };
+        //TODO: read tmp->velocities.
```

Comment:
You can just remove this line as you will use only functions which depend on the configuration only